projects/storefrontlib/cms-components/misc/icon/icon.component.ts

```diff
@@ -47,87 +47,100 @@ export class IconComponent {
+  @Input()
+  public set type(type: ICON_TYPE_STRING) {
+    // XXX this clashes with the type attribute of some html form elements, e.g. input or button
     this.setIcon(type);
   }
 
-  /**
-   * the icon provides an html fragment that is used to add SVG or text based icons.
-   */
-  icon: SafeHtml | undefined;
-
-  /**
-   * The `flip-at-rtl` class is added to the DOM for the style layer to flip the icon in RTL direction.
-   */
-  @HostBinding('class.flip-at-rtl') flipAtRtl: boolean;
-
-  /**
-   * The `flip-at-ltr` class is added to the DOM for the style layer to flip the icon in LTR direction.
-   */
-  @HostBinding('class.flip-at-ltr') flipAtLtr: boolean;
-
   /**
    * Maintains the applied style classes so we can remove them when the
    * icon type changes at run time.
    */
-  protected styleClasses: string[];
+  @HostBinding('class')
+  public styleClasses: string[] = [];
+
+  private iconValue?: string;
+  private iconResourceType: IconResourceType;
 
   constructor(
     protected iconLoader: IconLoaderService,
-    protected elementRef: ElementRef<HTMLElement>,
-    protected renderer: Renderer2
+    protected sanitizer: DomSanitizer
   ) {}
 
-  protected setIcon(type: ICON_TYPE): void {
+  protected setIcon(type: ICON_TYPE_STRING): void {
     if (!type || <string>type === '') {
       return;
     }
-    this.icon = this.iconLoader.getHtml(type);
-    this.addStyleClasses(type);
+    this.iconResourceType = this.iconLoader.getResourceType(type);
+    if (this.isSvgIcon()) {
+      // looks like angular does not sanitize URLs when binding to the xlink:href attribute of svg elements
+      // so we need to sanitze the URL manually
+      this.iconValue =
+        this.sanitizer.sanitize(
+          SecurityContext.URL,
+          this.iconLoader.getSvgPath(type) || null
+        ) || undefined;
+    } else if (this.isTextIcon()) {
+      this.iconValue = this.iconLoader.getSymbol(type);
+    } else {
+      // assume IconResourceType.LINK by default
+      this.iconValue = undefined;
+    }
     this.iconLoader.addLinkResource(type);
-    this.flipIcon(type);
+    this.addStyleClasses(type);
+  }
+
+  public get icon(): string | undefined {
+    return this.iconValue;
+  }
+
+  public isSvgIcon(): boolean {
+    return this.iconResourceType === IconResourceType.SVG;
+  }
+
+  public isTextIcon(): boolean {
+    return this.iconResourceType === IconResourceType.TEXT;
   }
 
   /**
-   * The icons supports flipping for some icons to support rtl and ltr directions.
+   * If necessary, adds CSS classes for flipping for some icons to support rtl and ltr directions.
    */
-  protected flipIcon(type: ICON_TYPE) {
-    // TODO: this can be dropped with the next major release.
+  protected flipIcon(type: ICON_TYPE_STRING) {
+    // TODO: this check can be dropped with the next major release.
     if (!this.iconLoader.getFlipDirection) {
       return;
     }
     const iconDirection = this.iconLoader.getFlipDirection(type);
```

Comment:
I don't know if in practice this could happen, but it seems that calling flipIcon() multiple times with different types can accumulate both direction classes in the array.  Could that be the case givent that styleClasses is a class attribute and classes are always added to it?

With the previous logic it looks like only one or the other could be applied.

Reply:
I think that Angular's `@HostBinding('class')` will ultimately use DOM's `classList` under the hood. this is a  [DOMTokenList](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList) which operates as a set.

Just call `document.body.classList.add('foo')` multiple times and observe the result.

But I'll see if any tests cover this. if not, I'll add new ones....